clang/lib/Sema/Sema.cpp

```diff
@@ -1621,6 +1621,18 @@ static void emitCallStackNotes(Sema &S, FunctionDecl *FD) {
   }
 }
 
+bool Sema::isSyclGlobalVariableAllowedType(QualType Ty) {
```

Comment:
@schittir, could you explain why we are checking Sema::isSyclDeviceGlobalType() and Sema::isSyclGlobalVariableAllowedType() in two different files? Sema vs SemaDecl.cpp file?

Reply:
Good point, @smanna12 
No specific reason other than that isSyclGlobalVariableAllowedType is used in a couple of files where as so far I've used isSyclDeviceGlobal only in SemaDecl. I put the definition local to where as I was using it. I will move them into one file.

Reply:
> Good point, @smanna12 No specific reason other than that isSyclGlobalVariableAllowedType is used in a couple of files where as so far I've used isSyclDeviceGlobal only in SemaDecl. I put the definition local to where as I was using it. I will move them into one file.

You can combine them into one function using template. Both are doing same but only different attribute name.

Reply:
Ok. Will do. Thank you!